Source/Task/AsyncQueue.cpp

```diff
@@ -245,21 +294,34 @@ class Queue
             isEmpty = m_queueHead.Flink == &m_queueHead;
             if (isEmpty)
             {
+#if HC_PLATFORM_IS_MICROSOFT
                 ResetEvent(m_event);
+#endif
             }
         }
 
-        if (isEmpty && m_dispatchMode == AsyncQueueDispatchMode_FixedThread &&
-            GetThreadId(m_apcThread) == GetCurrentThreadId())
+        if (isEmpty && m_dispatchMode == AsyncQueueDispatchMode_FixedThread)
```

Comment:
I think you can simplify this by making the entire block wrapped in the MS platform check.  There's no value in doing the first set of if statements and asserting for other platforms.  The whole block is just designed to pump out the user callbacks, which don't exist on other platforms.